modules/percolator/src/main/java/org/elasticsearch/percolator/PercolatorFieldMapper.java

```diff
@@ -405,4 +455,28 @@ static void verifyQuery(QueryBuilder queryBuilder) {
         }
     }
 
+    static byte[] encodeRange(String rangeFieldName, byte[] minEncoded, byte[] maxEncoded) {
+        assert minEncoded.length == maxEncoded.length;
+        byte[] bytes = new byte[BinaryRange.BYTES * 2];
+
+        // First compute hash for field name and write the full hash into the byte array
+        BytesRef fieldAsBytesRef = new BytesRef(rangeFieldName);
+        MurmurHash3.Hash128 hash = new MurmurHash3.Hash128();
+        MurmurHash3.hash128(fieldAsBytesRef.bytes, fieldAsBytesRef.offset, fieldAsBytesRef.length, 0, hash);
+        byte[] encodedH1 = new byte[Long.BYTES];
+        LongPoint.encodeDimension(hash.h1, encodedH1, 0);
+        byte[] encodedH2 = new byte[Long.BYTES];
+        LongPoint.encodeDimension(hash.h2, encodedH2, 0);
+        ByteBuffer bb = ByteBuffer.wrap(bytes);
+        bb.put(encodedH1).put(encodedH2).put(encodedH1).put(encodedH2);
```

Comment:
maybe we can simplify by calling `putLong(hash.h1)` directly and avoid the intermediate byte[]?